src/relations/backend_database.py

```diff
@@ -476,3 +487,17 @@ def check_backend(self) -> bool:
             self.charm.unit.status = WaitingStatus(wait_str)
             return False
         return True
+
+    def sync_hba(self, user: str) -> None:
+        """Wait for user to appear in pg_hba table."""
+        # Check for version supporting hardening
+        if self.backend_version and self.backend_version < "14.16":
+            return
```

Comment:
`14.15` is the current stable. Since it has no hardening, there's no point in waiting.